### ISSUE TITLE
Google signs out every time browser is closed

### ISSUE THREAD

**Issue by SuperWKM** [mozilla-mobile/firefox-ios#19909]
When I close out of the browser and search for something new, I am logged out of Google. If i refresh a couple of times it fixes.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-9012)

**Comment by afurlan-firefox** [comment]
@SuperWKM Thanks for reporting this. Can you provide more information? Are you using Normal Mode or Private Mode? If you can upload a video and more details, it will be helpful!

**Comment by SuperWKM** [comment]
> @SuperWKM Thanks for reporting this. Can you provide more information? Are you using Normal Mode or Private Mode? If you can upload a video and more details, it will be helpful!

Of course. I am using normal mode. Here is a video. As you can see, when I searched something I was not signed in to Google. After refreshing, dark mode turned on but I still wasn’t signed in. The second refresh, it signed me in. 

https://github.com/mozilla-mobile/firefox-ios/assets/167652214/91533662-aef8-48dc-af71-cf0801776216

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Andrei Bodea Diana Andreea Barladeanu can you check this issue? Thanks!

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Hello, we were able to reproduce this issue on v9000 (41381), with iPhone 15 (17.4).

Also, we confirm that sometimes, after refreshing, google dark theme turns on by itself.

!RPReplay_Final1714475142.mp4|width=590,height=1280,alt="RPReplay_Final1714475142.mp4"!

**Comment by data-sync-user** [comment]
➤ Daniel Dervishi commented:

Tested this on both Safari and Google Chrome. It seems to be an issue on Google’s end and not ours.

**Comment by data-sync-user** [comment]
➤ Daniel Dervishi commented:

I submitted a bug report related to this issue on Apple Feedback Assistant.

**Comment by data-sync-user** [comment]
➤ Daniel Dervishi commented:

Moved Into todo because the ticket was blocked as we need assistance from apple

**Comment by data-sync-user** [comment]
➤ Matt Reagan commented:

{quote}Tested this on both Safari and Google Chrome and the same behavior was present.{quote}

Daniel Dervishi This is interesting, I’m not able to replicate the problem in Safari or Brave. I can, however, replicate the bug in Firefox.

**Comment by data-sync-user** [comment]
➤ Matt Reagan commented:

Note: I also tried creating a simple iOS test app with a WebView and I’m not able to replicate the issue there, after quitting and relaunching gmail appears to be loaded correctly. I can, however, replicate reliably in

**Comment by data-sync-user** [comment]
➤ Daniel Dervishi commented:

https://mozilla-hub.atlassian.net/browse/FXIOS-9012?focusedCommentId=923066 ( https://mozilla-hub.atlassian.net/browse/FXIOS-9012?focusedCommentId=923066|smart-link ) 
Also no longer able to replicate on chrome or safari.
